### PR TITLE
Networking workarounds for AWS/GCE

### PR DESCRIPTION
**Pull request by tgross**
@misterbisson @max123 this PR includes changes to the makefile I've been using for orchestration to workaround problems I had with the AWS/GCE setup:
- use the host networking so that the IP the couchbase containers can see (and with which they register to consul) matches the host's IP
- fix the memory size of the nodes to fit within what we can get from AWS & GCE
- inject the consul IP into docker-compose via sed to a temp file rather than by trying (unsuccessfully!) to inject it via environment variables; I misread the docker-compose docs here.
- added a draft (as-yet-untested) digitalocean piece

### CODE REVIEW

File: docker-compose-swarm.yml [hunk 1]
```diff
@@ -11,15 +12,19 @@ consul:
...
 # Manually bootstrap the first instance, then...
 # Scale this tier and each additional container/instance will automatically
 # self-configure as a member of the cluster
 couchbase4:
     image: 0x74696d/triton-couchbase:4.0.0
+    net: host
     extra_hosts:
-    - consul:${CONSUL_IP}
-    mem_limit: 128g
+    - "consul:<CONSUL_IP>"
```
Comment (misterbisson) [inline docker-compose-swarm.yml:26]: Is the angle bracket syntax in `<CONSUL_IP>` significant to Swarm, or is that for a different tool?

File: start.bash [hunk 2]
```diff
@@ -28,7 +28,7 @@ while [ $COUCHBASERESPONSIVE != 1 ]; do
     RUNNING=$(docker inspect ${PREFIX}_couchbase${VERSION}_1 | json -a State.Running)
     if [ "$RUNNING" == "true" ]
     then
-        docker exec -it ${PREFIX}_couchbase${VERSION}_1 triton-bootstrap bootstrap benchmark
+        docker exec ${PREFIX}_couchbase${VERSION}_1 triton-bootstrap bootstrap benchmark
```
Comment (misterbisson) [inline start.bash:31]: Right. I should backport this change.

**Comment by misterbisson** [comment]
:pickles: to merge

File: docker-compose-swarm.yml (hunk 1 above)
Comment (tgross) [inline docker-compose-swarm.yml:26]: It's just to mark this for the line of sed here: https://github.com/misterbisson/clustered-couchbase-in-containers/pull/10/files#diff-edbb3d293f4a3df14cbfde338382c9dcR132

You can't pass environment variables from the client's host into docker-compose, so I'm injecting them via sed and then using that temp file as the input to docker-compose.

Comment (misterbisson) [inline docker-compose-swarm.yml:26]: Right, I saw that later on and should have noted it.

I'm begrudgingly thinking that's the best approach for a problem I've got.

Comment (tgross) [inline docker-compose-swarm.yml:26]: > begrudgingly

Yeah it's pretty awful but without some kind of name service you can't really get around it.

**Pull request merged**